internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -485,6 +569,28 @@ func updateDesiredState(ctx context.Context, req *runtimehooksv1.GeneratePatches
 		}
 	}
 
+	// Update the templates for all MachinePools.
+	for mpTopologyName, mp := range desired.MachinePools {
+		topologyName := requestTopologyName{mpTopologyName: mpTopologyName}
+		// Update the BootstrapTemplate.
```

Comment:
```suggestion
		// Update the BootstrapConfig.
```

This one is actually the BootstrapConfig of the Cluster (no other changes are needed here)